lib/std/sync.zig

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2021 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+
+const std = @import("std.zig");
+const root = @import("root");
+
+pub const atomic = @import("./sync/atomic.zig");
+pub const futex = @import("./sync/futex.zig");
+pub const primitives = @import("./sync/primitives.zig");
+
+pub usingnamespace if (@hasDecl(root, "sync"))
+    root.sync
+else if (std.builtin.single_threaded)
+    primitives.debug
```

Comment:
Why is this `debug` rather than `core`?

Reply:
core is for algorithms, the "Debug" impls are the "serial" ones. Unfortunate naming.